src/node.cc

```diff
@@ -383,7 +383,7 @@ MaybeLocal<Value> StartExecution(Environment* env, StartExecutionCallback cb) {
     return StartExecution(env, "internal/main/watch_mode");
   }
 
-  if (!first_argv.empty() && first_argv != "-") {
+  if (!first_argv.empty() && first_argv != "-" && first_argv != "/dev/stdin") {
```

Comment:
What if a file symlinks to /dev/stdin, or one of the fd 0 symlinks?

This approach doesn't account for those cases